data/abilities.ts

```diff
@@ -765,7 +765,11 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 				effect && effect.effectType === 'Move' &&
 				['mimikyu', 'mimikyutotem'].includes(target.species.id) && !target.transformed
 			) {
-				this.add('-activate', target, 'ability: Disguise');
+				if (this.gen === 7 && ["rollout", "iceball"].includes(effect.id)) {
```

Comment:
This behavior should be inside the mod/ directory for ``gen8dlc1`` and ``gen7``.